OTPKit/Features/TripPlanner/TripPlannerSheetView.swift

```diff
@@ -15,41 +15,130 @@ public struct TripPlannerSheetView: View {
 
     private func formatTimeDuration(_ duration: Int) -> String {
         if duration < 60 {
-            return "Total duration: \(duration) second\(duration > 1 ? "s" : "")"
+            return "\(duration) second\(duration > 1 ? "s" : "")"
         } else if duration < 3600 {
             let minutes = Double(duration) / 60
-            return String(format: "Total duration: %.1f minutes", minutes)
+            return String(format: "%.1f min", minutes)
         } else {
             let hours = Double(duration) / 3600
-            return String(format: "Total duration: %.1f hours", hours)
+            return String(format: "%.1f hours", hours)
         }
     }
 
     private func formatDistance(_ distance: Int) -> String {
         if distance < 1000 {
-            return "Total distance: \(distance) meters"
+            return "\(distance) meters"
         } else {
             let miles = Double(distance) / 1609.34
-            return String(format: "Total distance: %.1f miles", miles)
+            return String(format: "%.1f miles", miles)
         }
     }
 
+    private func formatBusSchedule(_ date: Date) -> String {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "h:mm a"
+        dateFormatter.amSymbol = "AM"
+        dateFormatter.pmSymbol = "PM"
+
+        let formattedTime = dateFormatter.string(from: date)
+
+        return "Bus scheduled at \(formattedTime)"
+    }
+
     public var body: some View {
         VStack {
             if let itineraries = locationManagerService.planResponse?.plan?.itineraries {
                 List(itineraries, id: \.self) { itinerary in
-                    let distance = itinerary.legs.map(\.distance).reduce(0, +)
                     Button(action: {
                         locationManagerService.selectedItinerary = itinerary
                         locationManagerService.planResponse = nil
                         dismiss()
                     }, label: {
-                        VStack(alignment: .leading) {
-                            Text(formatTimeDuration(itinerary.duration))
-                            Text(formatDistance(Int(distance)))
+                        HStack(spacing: 20) {
+                            VStack(alignment: .leading) {
+                                Text(formatTimeDuration(itinerary.duration))
+                                    .font(.title)
+                                    .fontWeight(.bold)
+                                    .foregroundStyle(.foreground)
+                                Text(formatBusSchedule(itinerary.startTime))
+                                    .foregroundStyle(.gray)
+
+                                FlowLayout {
+                                    ForEach(Array(zip(itinerary.legs.indices, itinerary.legs)), id: \.1) { index, leg in
+                                        switch leg.mode {
+                                        case "TRAM":
```

Comment:
It looks like these could be DRY'd up into a new view called `ItineraryLegView` or something like that. You pass in `leg.route`, a background color, and an image name into the initializer.

Reply:
Thanks for the feedback Aaron! I'll try it out